src/protocol/http/src/main/java/org/apache/jmeter/protocol/http/visualizers/RequestViewHTTP.java

```diff
@@ -287,10 +287,16 @@ public static Map<String, String[]> getQueryMap(String query) {
         String[] params = query.split(PARAM_CONCATENATE);
         for (String param : params) {
             String[] paramSplit = param.split("=");
-            String name = decodeQuery(paramSplit[0]);
+            String name = null;
+            boolean validNameAndValue = true;
```

Comment:
With the setting of `name` to `null` and conditionally setting it in case of a valid value, we can replace this variable `validNameAndValue` with a check for `name != null` (or more positive `name==null`).

But either way, I think the logic is correct now.

Reply:
I prefer keep the intermediate variable validNameAndValue, seems bring a better readness of the code (IMHO)